src/contracts/middleware/AvalancheL1Middleware.sol

```diff
@@ -812,8 +803,27 @@ contract AvalancheL1Middleware is IAvalancheL1Middleware, AssetClassRegistry {
         if (!balancerValidatorManager.isValidatorPendingWeightUpdate(validationID)) {
             revert AvalancheL1Middleware__WeightUpdateNotPending(validationID);
         }
-        // if the completeValidatorWeightUpdate fails, not sure if the previous bool is secure.
+
+        // Finish the weight update on the P-Chain
```

Comment:
Comment is not super acurate. Could be "Complete weight update upon P-Chain confirmation"